From b2a4efdb6ad7a53a681fc4d075c322bbafec6a41 Mon Sep 17 00:00:00 2001
From: Adithya R <gh0strider.2k18.reborn@gmail.com>
Date: Thu, 18 Aug 2022 21:29:34 +0530
Subject: [PATCH 6/6] core: sysprop: Write build display id to product prop

Some vendors override build id in odm or vendor, such as oplus.

Change-Id: I1c2de43ba7c3544710897ad9127d320dd02293fe
---
 core/sysprop.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/sysprop.mk b/core/sysprop.mk
index 44af367a38d..4adec48daa7 100644
--- a/core/sysprop.mk
+++ b/core/sysprop.mk
@@ -76,6 +76,9 @@ define generate-common-build-props
             echo "ro.$(1).product.cpu.abilist64=$(TARGET_CPU_ABI_LIST_64_BIT)" >> $(2);\
         )\
     )\
+    $(if $(filter product,$(1)),\
+        echo "ro.build.display.id=$(BUILD_DISPLAY_ID)" >> $(2);\
+    )\
     echo "ro.$(1).build.date=`$(DATE_FROM_FILE)`" >> $(2);\
     echo "ro.$(1).build.date.utc=`$(DATE_FROM_FILE) +%s`" >> $(2);\
     echo "ro.$(1).build.fingerprint=$(BUILD_FINGERPRINT_FROM_FILE)" >> $(2);\
-- 
2.34.1

